spikeinterface/core/tests/test_basesnippets.py

```diff
@@ -0,0 +1,108 @@
+from spikeinterface.core.waveform_tools import extract_waveforms_to_buffers
+from spikeinterface.core.numpysnippetsextractor import NumpySnippetsExtractor
+
+
+def test_BaseSnippets():
+    duration = 60
+    num_channels = 3
+    nbefore = 20
+    nafter = 44
+    recording = generate_recording(durations=[duration], num_channels=num_channels)
+    probe = recording.get_probe()
+    sorting = generate_sorting(durations=[duration], num_units=5)
+    strains = sorting.get_all_spike_trains()
+    peaks_times = np.sort(strains[0][0])
+    peak_dtype = [('sample_ind', 'int64'), ('unit_ind', 'int64'), ('segment_ind', 'int64')]
+    peaks2 = np.zeros(len(peaks_times), dtype=peak_dtype)
+    peaks2['sample_ind'] = peaks_times
+    peaks2['segment_ind'] = 0
+    peaks2['unit_ind'] = 0
+    wfs_arrays = extract_waveforms_to_buffers(recording, peaks2, [0], nbefore, nafter,
+                                    mode='shared_memory', return_scaled=False, folder=None, dtype=recording.get_dtype(),
+                                    sparsity_mask=None,n_jobs=1)
+    wfs = wfs_arrays[0][0] #extract class zero
+
+    nse = NumpySnippetsExtractor(wfs, peaks2['sample_ind'], recording.get_sampling_frequency(), 
+                                 nafter=nbefore, channel_ids=None)
+    nse = nse.set_probe(probe)
+
+    assert nse.get_probe() is not None
+    assert nse.get_num_segments() == 1
+
+    assert np.all(nse.ids_to_indices([0, 1, 2]) == [0, 1, 2])
+    assert np.all(nse.ids_to_indices(
+        [0, 1, 2], prefer_slice=True) == slice(0, 3, None))
+
+    # annotations / properties
+    nse.annotate(gre='ta')
+    assert nse.get_annotation('gre') == 'ta'
+
+    nse.set_channel_groups([0, 0, 1])
+    groups = nse.get_channel_groups()
+    assert np.array_equal(groups, [0, 0, 1])
+
+    nse.set_property('quality', [1., 3.3, np.nan])
+    values = nse.get_property('quality')
+    assert np.all(values[:2] == [1., 3.3, ])
+
+    # missing property
+    nse.set_property('string_property', ["ciao", "bello"], ids=[0, 1])
+    values = nse.get_property('string_property')
+    assert values[2] == ""
+
+    # setting an different type raises an error
+    assert_raises(Exception, nse.set_property, key='string_property_nan', values=["ciao", "bello"], ids=[0, 1],
+                  missing_value=np.nan)
+
+    # int properties without missing values raise an error
+    assert_raises(Exception, nse.set_property,
+                  key='int_property', values=[5, 6], ids=[1, 2])
+
+    nse.set_property('int_property', [5, 6], ids=[1, 2], missing_value=200)
+    values = nse.get_property('int_property')
+    assert values.dtype.kind == "i"
+
+    times0 = nse.get_frames(segment_index=0)
+    assert all(peaks2['sample_ind']==times0)
+    # TODO: after making another extractor
+    # dump/load dict
+    # dump/load json
+    # dump/load pickle
+    # cache to binary
+    #load from folder
+    # cache to memory
+    # cache joblib several jobs
+    # set/get Probe
+
```

Comment:
testing the dump / load is important feature of recording/sorting, we need to make some test also here.